### ISSUE TITLE
[Bug]: `frameElement()` does not work for frames in (closed) shadow root

### ISSUE THREAD

**Issue by Niek** [puppeteer/puppeteer#13155]
### Minimal, reproducible example

```TypeScript
import puppeteer from 'puppeteer';

(async () => {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();

  // Set test content with a frame inside a closed shadow DOM.
  await page.setContent(`
    <div id="shadow-host"></div>
    <script>
      const host = document.getElementById('shadow-host');
      const shadowRoot = host.attachShadow({ mode: 'closed' });
      const frame = document.createElement('iframe');
      frame.srcdoc = '<p>Inside frame</p>';
      shadowRoot.appendChild(frame);
    </script>
  `);

  // Attempt to select the frame on the page.
  const frame = page.frames()[0];
  const frameElement = await frame.frameElement();

  if (frameElement === null) {
    console.log('Frame element is null');
  } else {
    console.log('Frame element found');
  }

  await browser.close();
})();
```

### Background

It's not possible to get a frame element handle that is inside a closed shadow root.

### Expectation

It should be possible to get the element handle, like it's possible in playwright.

### Reality

Puppeteer does not seem to be able to get the right handle.

### Puppeteer version

23.4.1

### Node version

22.9.0

### Package manager

npm

### Package manager version

10.8.3

### Operating system

macOS

**Comment by OrKoN** [comment]
`page.frames()[0]` refers to the main frame, I assume `page.frames()[1]` was meant?

**Comment by Niek** [comment]
@OrKoN Absolutely right - my bad. Thanks for the fix in #13156!
